base/logger.go

```diff
@@ -34,3 +40,42 @@ func logCollationWorker(collateBuffer chan string, logger *log.Logger, bufferSiz
 		}
 	}
 }
+
+// runLogDeletion will delete rotated logs for the supplied logLevel. It will only perform these deletions when the
+// cumulative size of the logs are above the supplied sizeLimitMB.
+// logDirectory is the supplied directory where the logs are stored.
```

Comment:
Add unit tests for runLogDeletion to verify handling for scenarios:
- standard case
- hit low watermark but not high watermark
- single file triggering low + high watermark
- no deletion required (didn't hit low watermark)
- file datetime boundary conditions (end of month, end of year, etc)